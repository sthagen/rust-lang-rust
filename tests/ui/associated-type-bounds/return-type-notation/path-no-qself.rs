#![feature(return_type_notation)]
//~^ WARN the feature `return_type_notation` is incomplete

trait Trait {
    fn method() -> impl Sized;
}

fn test()
where
    Trait::method(..): Send,
    //~^ ERROR ambiguous associated type
{
}

fn main() {}
