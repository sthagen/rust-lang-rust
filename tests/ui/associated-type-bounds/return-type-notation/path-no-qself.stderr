warning: the feature `return_type_notation` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/path-no-qself.rs:1:12
   |
LL | #![feature(return_type_notation)]
   |            ^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #109417 <https://github.com/rust-lang/rust/issues/109417> for more information
   = note: `#[warn(incomplete_features)]` on by default

error[E0223]: ambiguous associated type
  --> $DIR/path-no-qself.rs:10:5
   |
LL |     Trait::method(..): Send,
   |     ^^^^^^^^^^^^^^^^^
   |
help: if there were a type named `Example` that implemented `Trait`, you could use the fully-qualified path
   |
LL |     <Example as Trait>::method: Send,
   |     ~~~~~~~~~~~~~~~~~~~~~~~~~~

error: aborting due to 1 previous error; 1 warning emitted

For more information about this error, try `rustc --explain E0223`.
